pandas/_libs/join.pyx

```diff
@@ -309,8 +307,8 @@ left_join_indexer_unique_int64 = left_join_indexer_unique["int64_t"]
 left_join_indexer_unique_uint64 = left_join_indexer_unique["uint64_t"]
 
 
-# @cython.wraparound(False)
-# @cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.boundscheck(False)
```

Comment:
Looking at the git history my impression is that this may have been commented-out unintentionally.  Maybe the CI will tell us otherwise.